gene2phenotype_project/gene2phenotype_app/views.py

```diff
@@ -898,3 +898,39 @@ def update(self, request, *args, **kwargs):
 
         else:
             return Response({"message": "Failed to update data", "details": serializer.errors})
+
+class PublishRecord(APIView):
+    http_method_names = ['post', 'head']
+    serializer_class = CurationDataSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def post(self, request, stable_id, *args, **kwargs):
+        user = self.request.user
+
+        try:
+            # Get curation record
+            curation_obj = CurationData.objects.get(stable_id__stable_id=stable_id,
+                                                    user__email=user)
+
+            # Check if there is enough data to publish the record
+            locus_obj = self.serializer_class().validate_to_publish(curation_obj)
+
+            # Publish record
+            try:
+                lgd_obj = self.serializer_class(context={'user': user}).publish(curation_obj)
+                # Delete entry from 'curation_data'
+                curation_obj.delete()
+
+                return Response({
+                    "message": f"Record '{lgd_obj.stable_id.stable_id}' published succesfully"
+                    }, status=status.HTTP_201_CREATED)
+
+            except LocusGenotypeDisease.DoesNotExist:
+                Response({
+                    "message": f"Failed to publish record ID '{stable_id}'"
+                    }, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
if the LGD entry does not exist, should we return 404 not found error?

Reply:
This except is thrown if the entry could not be published i.e. if the new LocusGenotypeDisease could not be created (meaning it doesn't exist) - this happens because there was some kind of internal error.